File: jobs_package/Jobs_Flutter_addon/lib/src/models/cover_letter.dart
class CoverLetter {
  const CoverLetter({
    this.id,
    required this.candidateId,
    required this.title,
    required this.body,
  });

  final int? id;
  final int candidateId;
  final String title;
  final String body;

  factory CoverLetter.fromJson(Map<String, dynamic> json) => CoverLetter(
        id: json['id'] as int?,
        candidateId: json['candidate_id'] as int,
        title: json['title'] as String,
        body: json['body'] as String? ?? '',
      );

  Map<String, dynamic> toJson() => {
        if (id != null) 'id': id,
        'candidate_id': candidateId,
        'title': title,
        'body': body,
      };
}

File: jobs_package/Jobs_Laravel_package/src/Models/CoverLetter.php
<?php

namespace Jobs\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class CoverLetter extends Model
{
    use HasFactory;

    protected $fillable = [
        'candidate_id',
        'title',
        'body',
    ];
}
